### PR TITLE
Chore: implement yarn workspaces

### PR DESCRIPTION
**Pull request by gui-santos**
# Purpose of PR

This PR implements [yarn workspaces](https://classic.yarnpkg.com/en/package/workspaces) to our monorepo!
Basically, it hoists our dependencies to one node_modules folder at the root of the repo with only one yarn.lock and then it creates symlinks to those dependencies in our packages. This will make us avoid errors like the one on the website about having two versions of react

Please take a careful look at this PR since it changes a lot of stuff and it might explode something

I recommend doing this before checking it locally:

1. `nvm use`
2. `yarn lerna clean` to remove node_modules from packages
3. `yarn rimraf node_modules` to remove node_modules from root (I added rimraf as a dev dep but we can remove it)
4. `yarn`

## PR Checklist

- [ ] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [ ] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

File: packages/components/forma-36-react-timepicker/src/Timepicker.tsx [hunk 1]
```diff
@@ -79,7 +79,7 @@ function createHours() {
     hours.push(addMinutes(addHours(startOfDay(new Date()), hour), 30));
   }
   return orderBy(hours, (time: Date) => time, 'asc').map((m: Date) =>
-    format(m, DATEFNS_12H_FORMAT)
+    format(m, DATEFNS_12H_FORMAT),
```
Comment (gui-santos) [inline packages/components/forma-36-react-timepicker/src/Timepicker.tsx:82]: The changes in this file were done by prettier

`tsdx lint` was complaining about its formatting

File: packages/forma-36-fcss/scss/styles.scss [hunk 2]
```diff
@@ -1,4 +1,4 @@
-@import '../node_modules/@contentful/forma-36-tokens/dist/scss/index.scss';
+@import '../../../node_modules/@contentful/forma-36-tokens/dist/scss/index.scss';
```
Comment (gui-santos) [inline packages/forma-36-fcss/scss/styles.scss:1]: Because yarn workspaces moves our dependencies to the root I had to do this

I tried to do something like `@contentful/forma-36-tokens/dist/scss/index.scss`, but for this to work something is still missing
we could do it in another PR I think

File: packages/forma-36-website/src/components/Layout.css [hunk 3]
```diff
@@ -1,4 +1,4 @@
-@import './../../node_modules/@contentful/forma-36-tokens/dist/css/index.css';
+@import '../../../../node_modules/@contentful/forma-36-tokens/dist/css/index.css';
```
Comment (gui-santos) [inline packages/forma-36-website/src/components/Layout.css:1]: Same comment as the other import

**Review by denkristoffer — APPROVED** [review]

**Review by burakukula — APPROVED** [review]
Looks good to me 🤝

File: .prettierignore [hunk 4]
```diff
@@ -1,3 +1,2 @@
-package.json
```
Comment (mshaaban0) [inline .prettierignore:1]: 👍

File: packages/components/forma-36-react-timepicker/package.json [hunk 5]
```diff
@@ -16,19 +16,14 @@
     "lint": "tsdx lint"
   },
   "peerDependencies": {
-    "react": ">=16"
+    "react": "^16.8.6",
```
Comment (mshaaban0) [inline packages/components/forma-36-react-timepicker/package.json:19]: Do we have to pin the peer version here? isn't that what is causing the duplicate react versions?

Comment (gui-santos) [reply packages/components/forma-36-react-timepicker/package.json:19]: I thought the duplication was caused by having it in devDependencies and here
but this was happening only on the website, no?

File: packages/components/forma-36-react-datepicker/tsconfig.json [hunk 6]
```diff
@@ -20,8 +20,9 @@
     "noImplicitReturns": true,
     "noFallthroughCasesInSwitch": true,
     "moduleResolution": "node",
+    "allowJs": true,
     "baseUrl": "./",
-    "typeRoots": ["node_modules/@types", "./typings.d.ts"],
+    "typeRoots": ["../../../node_modules/@types", "./typings.d.ts"],
```
Comment (mshaaban0) [inline packages/components/forma-36-react-datepicker/tsconfig.json:25]: So this now listens to the global types only? what if someone adds a package only for this component, will the types be available?

File: packages/components/forma-36-react-timepicker/package.json (hunk 5 above)
Comment (mshaaban0) [reply packages/components/forma-36-react-timepicker/package.json:19]: yes, but `peerDependencies` is different, it's used only for consumer packages, it tells the consumer package i.e. "user_interface" which version of react it needs. it's not used for compilation

Comment (gui-santos) [reply packages/components/forma-36-react-timepicker/package.json:19]: So you are suggesting to keep devDep and removing peerDependencies totally, is that it?
Don't this mean that someone consuming react-timepicker that has no react will get an error?

Comment (mshaaban0) [reply packages/components/forma-36-react-timepicker/package.json:19]: `peerDep` should stay, but with `>=16` just without pinning the version to avoid duplicate installation if consumer has `16.9.0` for example.
As for `devDep` i think we should keep it removed yes, should be able to get the react version from root I think.

File: packages/components/forma-36-react-datepicker/tsconfig.json (hunk 6 above)
Comment (gui-santos) [reply packages/components/forma-36-react-datepicker/tsconfig.json:25]: I'm not sure why we need this tbh, but I need this to get the types because the compiler was complaining and I want to focus on one refactor at a time
I even thought about unifying the tsconfigs in a later PR

What I can say to you is that when adding a dependency for only one package this dependency will be added in the local node_modules, but a dependency used by more than a package is hoisted to the root by yarn-workspaces

if know more about this `typeRoots`, pls tell me :) I can use help

File: packages/components/forma-36-react-timepicker/package.json (hunk 5 above)
Comment (gui-santos) [reply packages/components/forma-36-react-timepicker/package.json:19]: Ahhh I got it! I will make the change

Comment (gui-santos) [reply packages/components/forma-36-react-timepicker/package.json:19]: Thanks a lot 🖤

File: packages/components/forma-36-react-datepicker/tsconfig.json (hunk 6 above)
Comment (mshaaban0) [reply packages/components/forma-36-react-datepicker/tsconfig.json:25]: Having a base `tsconfig` file is ideal, and then we can extend it for packages if needed. Let's pair on this tomorrow, and get it cleaned up 👍

File: packages/components/forma-36-react-timepicker/package.json (hunk 5 above)
Comment (denkristoffer) [reply packages/components/forma-36-react-timepicker/package.json:19]: It should still be `>=16.8` because of hooks.

File: packages/forma-36-fcss/scss/styles.scss (hunk 2 above)
Comment (m10l) [reply packages/forma-36-fcss/scss/styles.scss:1]: Hmm, worried this change will break things when using the package in another project

Comment (denkristoffer) [reply packages/forma-36-fcss/scss/styles.scss:1]: I think you're right @m10l, but this was brittle before anyway because of the relative path. We should find out how to get the absolute import working.

Comment (gui-santos) [reply packages/forma-36-fcss/scss/styles.scss:1]: I will try more!
I suspect that we need to tell webpack or whatever is building the packages that `@contentful/forma-36-tokens` is a real dependency in node_modules in the root

And again, my change is just updating the relative path. The version in production is using a relative path already

File: packages/components/forma-36-react-datepicker/tsconfig.json (hunk 6 above)
Comment (gui-santos) [reply packages/components/forma-36-react-datepicker/tsconfig.json:25]: I totally removed this typeRoots and everything seems to be fine actually 👀 

or it will explode something that I'm missing

File: packages/forma-36-fcss/scss/styles.scss (hunk 2 above)
Comment (gui-santos) [reply packages/forma-36-fcss/scss/styles.scss:1]: Hey! I managed to make the relative path work, but I needed to add this dependency
https://www.npmjs.com/package/node-sass-import

And to ask yarn to not hoist it because it's not working properly when hoisted

And then we use it in the npm-script `build` like this:
`node-sass scss/styles.scss -o dist --output-style compressed --importer node_modules/node-sass-import`

File: packages/components/forma-36-react-datepicker/tsconfig.json [hunk 7]
```diff
@@ -20,8 +20,8 @@
     "noImplicitReturns": true,
     "noFallthroughCasesInSwitch": true,
     "moduleResolution": "node",
+    "allowJs": true,
```
Comment (denkristoffer) [inline packages/components/forma-36-react-datepicker/tsconfig.json:23]: Is there a reason to add this?

File: packages/components/forma-36-react-timepicker/jest.config.js [hunk 8]
```diff
@@ -1,5 +0,0 @@
-module.exports = {
```
Comment (denkristoffer) [inline packages/components/forma-36-react-timepicker/jest.config.js:1]: Nice to get rid of this again 😍

File: packages/components/forma-36-react-timepicker/package.json [hunk 9]
```diff
@@ -39,9 +34,8 @@
     "@types/lodash.orderby": "^4.6.6",
     "@types/react": "^16.9.11",
     "@types/react-dom": "^16.9.3",
+    "eslint-plugin-prettier": "^3.1.4",
```
Comment (denkristoffer) [inline packages/components/forma-36-react-timepicker/package.json:37]: Is this needed?

File: packages/components/forma-36-react-timepicker/tsconfig.json [hunk 10]
```diff
@@ -20,6 +20,7 @@
     "noImplicitReturns": true,
     "noFallthroughCasesInSwitch": true,
     "moduleResolution": "node",
+    "allowJs": true,
```
Comment (denkristoffer) [inline packages/components/forma-36-react-timepicker/tsconfig.json:23]: As the other comment, is there a reason to allow JS now?

File: packages/forma-36-react-components/package.json [hunk 11]
```diff
@@ -30,14 +30,14 @@
     "@storybook/addon-essentials": "^6.0.21",
     "@storybook/addon-knobs": "^6.0.21",
     "@storybook/addon-storysource": "^6.0.21",
-    "@storybook/react": "6.1.0-alpha.10",
+    "@storybook/react": "6.0.27",
```
Comment (denkristoffer) [inline packages/forma-36-react-components/package.json:33]: Why the downgrade?

File: packages/forma-36-react-components/package.json [hunk 12]
```diff
@@ -97,7 +97,7 @@
     "storybook-prop-types-addon": "1.0.2",
     "style-loader": "^0.23.1",
     "travis-deploy-once": "^4.4.1",
-    "typescript": "3.7.5",
+    "typescript": "3.9.7",
```
Comment (denkristoffer) [inline packages/forma-36-react-components/package.json:100]: 😍

File: packages/forma-36-website/src/components/Layout.css [hunk 13]
```diff
@@ -1,4 +1,4 @@
-@import './../../node_modules/@contentful/forma-36-tokens/dist/css/index.css';
+@import '~@contentful/forma-36-tokens/dist/css/index.css';
```
Comment (denkristoffer) [inline packages/forma-36-website/src/components/Layout.css:1]: I guess no one else actually uses this package, but in theory, if they did, would this import work for them? Shouldn't it be absolute?

**Review by denkristoffer — CHANGES_REQUESTED** [review]
Okay, I added some annoying questions, sorry 😬

File: packages/forma-36-website/src/components/Layout.css (hunk 13 above)
Comment (mshaaban0) [reply packages/forma-36-website/src/components/Layout.css:1]: unfortunately, that works with webpack, we're using `node-sass` directly. we should revisit the build system for `fcss` packages later, or even deprecate it, I remember Johannes mentions multiple times it's not being used much.

File: packages/components/forma-36-react-datepicker/package.json [hunk 14]
```diff
@@ -40,8 +41,6 @@
     "@types/react-dom": "^16.9.3",
```
Comment (mshaaban0) [inline packages/components/forma-36-react-datepicker/package.json:41]: Should we remove the `@types` as well?

File: packages/components/forma-36-react-timepicker/package.json [hunk 15]
```diff
@@ -16,19 +16,14 @@
     "lint": "tsdx lint"
   },
   "peerDependencies": {
-    "react": ">=16"
+    "react": ">=16.8",
+    "react-dom": ">=16.8"
```
Comment (mshaaban0) [inline packages/components/forma-36-react-timepicker/package.json:20]: Do we only support `16.8` or anything above `16`?

File: packages/forma-36-fcss/scss/styles.scss [hunk 16]
```diff
@@ -1,4 +1,4 @@
-@import '../node_modules/@contentful/forma-36-tokens/dist/scss/index.scss';
+@import '@contentful/forma-36-tokens/dist/scss/index.scss';
```
Comment (mshaaban0) [inline packages/forma-36-fcss/scss/styles.scss:1]: Aha so you used the `node-sass-importer` Noice 👍

File: packages/forma-36-react-components/package.json [hunk 17]
```diff
@@ -170,7 +170,9 @@
     "extend": "3.0.2",
     "@types/react": "16.8.6",
     "set-value": "2.0.1",
-    "js-yaml": "3.13.1"
+    "js-yaml": "3.13.1",
+    "react": "^16.8.6",
+    "react-dom": "^16.8.6"
```
Comment (mshaaban0) [inline packages/forma-36-react-components/package.json:175]: Do we need this to be in resolutions, [when to use resolutions](https://classic.yarnpkg.com/en/docs/selective-version-resolutions/#toc-why-would-you-want-to-do-this)?

File: packages/components/forma-36-react-datepicker/tsconfig.json (hunk 6 above)
Comment (mshaaban0) [reply packages/components/forma-36-react-datepicker/tsconfig.json]: @gui-santos sounds good, if the build is passing then it's reading it properly 👍 
We will look into unifying the tsconfig in another PR <3

File: packages/forma-36-react-components/package.json (hunk 17 above)
Comment (denkristoffer) [reply packages/forma-36-react-components/package.json:175]: IMO we should avoid resolutions as much as possible, as they can make it difficult to understand the actual version of a package that's used. The version in a `package.json` might not actually be the version used. Also, as far as I remember they don't get updated by tools like Dependabot.

File: packages/components/forma-36-react-datepicker/tsconfig.json (hunk 7 above)
Comment (gui-santos) [reply packages/components/forma-36-react-datepicker/tsconfig.json:23]: Oh I forgot now, but I got an error when building the datepicker and the error message suggested to set `allowJs` to true
I will remove this and get the error again so we can analyze this better because I forgot hahahah

File: packages/components/forma-36-react-timepicker/package.json (hunk 9 above)
Comment (gui-santos) [reply packages/components/forma-36-react-timepicker/package.json:37]: When I was linting (or building) the time picker `tsdx` telling me that `eslint-plugin-prettier` could not be found, but this was before I learned about the `nohoist` config so maybe we can use that instead

What I can say is that apparently `tsdx` does not understand workspaces well right now

I will post the error we get here so we can all see. And I will try to find the open issue in `tsdx` related to the error
maybe you all can help me with alternatives 🙌

File: packages/components/forma-36-react-timepicker/tsconfig.json (hunk 10 above)
Comment (gui-santos) [reply packages/components/forma-36-react-timepicker/tsconfig.json:23]: The same thing that happened to date picker happened here too, let me get the error and post it here

File: packages/forma-36-react-components/package.json (hunk 11 above)
Comment (gui-santos) [reply packages/forma-36-react-components/package.json:33]: I think when I was implementing popper.js for the first time I had to upgrade to the alpha version for some reason

And then now I downgraded to the last stable version to check if we really needed alpha and everything seems to work fine! So I thought we would prefer to have the stable version of storybook

Wdyt? We can go back to alpha

File: packages/forma-36-react-components/package.json (hunk 12 above)
Comment (gui-santos) [reply packages/forma-36-react-components/package.json:100]: I tried to update to v4 but I got too many errors so let's keep it for the next time 😄

File: packages/forma-36-website/src/components/Layout.css (hunk 13 above)
Comment (gui-santos) [reply packages/forma-36-website/src/components/Layout.css:1]: Uhmm I don't know now
`@contentful/forma-36-tokens` is a dependency in this package (website)
and I thought that `~` just tells webpack (well gatsby using webpack) to get this in node_modules
isn't it how this works?

File: packages/components/forma-36-react-datepicker/package.json (hunk 14 above)
Comment (gui-santos) [reply packages/components/forma-36-react-datepicker/package.json:41]: Hmm I don't know, I will remove it and if it does not break we remove it for real hahahahah

File: packages/components/forma-36-react-timepicker/package.json (hunk 15 above)
Comment (gui-santos) [reply packages/components/forma-36-react-timepicker/package.json:20]: I thought because we use hooks we want 16.8

Comment (gui-santos) [reply packages/components/forma-36-react-timepicker/package.json:20]: ahhh wait these components don't use hooks, right?

File: packages/forma-36-react-components/package.json (hunk 17 above)
Comment (gui-santos) [reply packages/forma-36-react-components/package.json:175]: That's why we have code reviews!
I think this is a leftover from some stuff that I was trying to do
I will remove it, thanks for spotting it 🙌

File: packages/forma-36-website/src/components/Layout.css (hunk 13 above)
Comment (mshaaban0) [reply packages/forma-36-website/src/components/Layout.css:1]: Ah yeah, it's the website, it should be fine. I thought it's the `fcss` package, sorry 🤦

File: packages/forma-36-react-components/package.json (hunk 11 above)
Comment (mshaaban0) [reply packages/forma-36-react-components/package.json:33]: for me stable is good if it works 👍

File: packages/forma-36-website/src/components/Layout.css (hunk 13 above)
Comment (gui-santos) [reply packages/forma-36-website/src/components/Layout.css:1]: @mshaaban0 your point about the `fcss` is still a valid point tough
I think `tokens` pkg is using it, but maybe we could avoid it

File: packages/components/forma-36-react-timepicker/package.json (hunk 15 above)
Comment (mshaaban0) [reply packages/components/forma-36-react-timepicker/package.json:20]: Good point about the hooks, I thought it was much earlier than `16.8` 
we can keep it for consistency or fallback to `16` if we want to be accurate, up to you.

File: packages/components/forma-36-react-datepicker/tsconfig.json (hunk 7 above)
Comment (gui-santos) [reply packages/components/forma-36-react-datepicker/tsconfig.json:23]: Good news! I removed this and no errors! hahahaha

File: packages/components/forma-36-react-timepicker/package.json (hunk 9 above)
Comment (gui-santos) [reply packages/components/forma-36-react-timepicker/package.json:37]: So this is what we get when I remove this and run `tsdx lint`:
```
Error: Failed to load plugin 'prettier' declared in 'BaseConfig': Cannot find module 'eslint-plugin-prettier'
Require stack:
- /Users/gui.santos/Contentful/forma-36/packages/components/forma-36-react-timepicker/__placeholder__.js
Referenced from: BaseConfig
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.resolve (internal/modules/cjs/helpers.js:33:19)
    at Object.resolve (/Users/gui.santos/Contentful/forma-36/node_modules/eslint/lib/shared/relative-module-resolver.js:44:50)
    at ConfigArrayFactory._loadPlugin (/Users/gui.santos/Contentful/forma-36/node_modules/eslint/lib/cli-engine/config-array-factory.js:959:39)
    at ConfigArrayFactory._loadExtendedPluginConfig (/Users/gui.santos/Contentful/forma-36/node_modules/eslint/lib/cli-engine/config-array-factory.js:779:29)
    at ConfigArrayFactory._loadExtends (/Users/gui.santos/Contentful/forma-36/node_modules/eslint/lib/cliengine/config-array-factory.js:725:29)
    at ConfigArrayFactory._normalizeObjectConfigDataBody (/Users/gui.santos/Contentful/forma36/node_modules/eslint/lib/cli-engine/config-array-factory.js:660:25)
    at _normalizeObjectConfigDataBody.next (<anonymous>)
    at ConfigArrayFactory._normalizeObjectConfigData (/Users/gui.santos/Contentful/forma36/node_modules/eslint/lib/cli-engine/config-array-factory.js:596:20)
    at _normalizeObjectConfigData.next (<anonymous>)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Does anyone have any clue?

Comment (denkristoffer) [reply packages/components/forma-36-react-timepicker/package.json:37]: tsdx should install `eslint-plugin-prettier` for us. Is it being hoisted or did you add it to `nohoist`?

Comment (gui-santos) [reply packages/components/forma-36-react-timepicker/package.json:37]: I'm getting it even when I do this:

```
"nohoist": [
      "@contentful/forma-36-fcss/node-sass-import",
      "**/eslint-plugin-prettier",
      "**/eslint-plugin-prettier/**"
    ]
```

Should I put tsdx in this list instead? It's only used in timepicker and date picker right now

Comment (gui-santos) [reply packages/components/forma-36-react-timepicker/package.json:37]: yep! I think not hoisting tsdx worked

I have to say, this is a con for tsdx. It should work with workspaces

Comment (denkristoffer) [reply packages/components/forma-36-react-timepicker/package.json]: There's this, but doesn't seem like there's much actual information there: https://github.com/formium/tsdx/issues/275

Comment (gui-santos) [reply packages/components/forma-36-react-timepicker/package.json]: Yep! I saw this issue before. I think it's ok to not hoist tsdx for now. It's used only in date and timepicker

wdyt?

Comment (denkristoffer) [reply packages/components/forma-36-react-timepicker/package.json]: I agree, but I wish I understood why it's giving us problems 🙂 [Formik is a monorepo](https://github.com/formium/formik) using tsdx as well and they don't have to jump through hoops to get it working. Maybe it'll get better once `react-components` also use tsdx.

**Review by denkristoffer — APPROVED** [review]
LGTM!

File: packages/components/forma-36-react-timepicker/package.json (hunk 9 above)
Comment (gui-santos) [reply packages/components/forma-36-react-timepicker/package.json]: Let's keep this in mind and try to go back to an ideal setup in the next PR's 🙌

File: package.json [hunk 18]
```diff
@@ -2,14 +2,26 @@
...
   "private": true,
+  "workspaces": {
+    "packages": [
+      "packages/*",
+      "packages/components/*"
+    ],
+    "nohoist": [
+      "@contentful/forma-36-fcss/node-sass-import",
+      "**/tsdx",
+      "**/tsdx/**"
```
Comment (mshaaban0) [inline package.json:13]: Just a question, why we need this to be here?

**Review by mshaaban0 — APPROVED** [review]
Nice work Gui ✨👏

File: package.json (hunk 18 above)
Comment (gui-santos) [reply package.json:13]: Because of @denkristoffer about the eslint-prettier pkg

File: packages/components/forma-36-react-timepicker/package.json (hunk 9 above)
Comment (gui-santos) [reply packages/components/forma-36-react-timepicker/package.json]: @mshaaban0 this is why we need tsdx to not be hoisted

File: package.json (hunk 18 above)
Comment (mshaaban0) [reply package.json:13]: Cool, thank you
